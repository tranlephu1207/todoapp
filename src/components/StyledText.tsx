import { Text, TextProps } from 'react-native';

interface StyledTextProps extends TextProps {
  font?: string;
}

export default function StyledText(props: StyledTextProps) {
  return (
    <Text {...props} style={[props.style, !!props.font && { fontFamily: props.font }]} />
  );
}
