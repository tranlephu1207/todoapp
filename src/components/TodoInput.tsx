import { Dimensions, StyleSheet, TextInput, TouchableOpacity, View } from 'react-native';

import Icon from '@expo/vector-icons/Feather';
import React from 'react';

const { width: windowWidth } = Dimensions.get('window');

const styles = StyleSheet.create({
  textInputContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomColor: 'rgb(222, 222, 222)',
    borderBottomWidth: 1,
    paddingHorizontal: 16,
    paddingBottom: 5,
    width: windowWidth,
  },
  textInput: {
    flex: 1,
    fontSize: 22,
    fontWeight: 'bold',
    color: 'white',
    maxHeight: windowWidth * 0.5,
    flexShrink: 1,
  }
});

interface TodoInput {
  value?: string;
  onChangeText?: (text: string) => void;
  onPressButton?: () => void;
  buttonIcon: React.ReactElement<typeof Icon>;
}

const TodoInput: React.FC<TodoInput> = ({
  value,
  onChangeText,
  onPressButton,
  buttonIcon
}) => {
  return (
    <View style={styles.textInputContainer}>
      <TextInput
        style={styles.textInput}
        multiline
        blurOnSubmit={true}
        placeholderTextColor="white"
        placeholder="Do it now!"
        value={value}
        onChangeText={onChangeText}
      />
      <TouchableOpacity
        onPress={onPressButton}
        style={{ marginLeft: 15 }}
      >
        {buttonIcon}
      </TouchableOpacity>
    </View>
  );
};

export default TodoInput;