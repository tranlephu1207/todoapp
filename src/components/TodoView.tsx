import { StyleSheet, TouchableOpacity, View } from 'react-native';

import Icon from '@expo/vector-icons/Feather';
import React from 'react';
import { Text } from '@components';
import { Todo } from '@features/todo/types.d';
import { useCallback } from 'react';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderColor: '#FFFFFF',
    borderBottomWidth: 1.5,
    width: '100%',
    alignItems: 'center',
    minHeight: 40,
    paddingVertical: 10,
  },
  rightIcon: {
    marginRight: 15,
    marginLeft: 'auto'
  },
  leftIcon: {
    marginLeft: 15
  },
  title: {
    paddingHorizontal: 10,
    fontSize: 17,
    fontWeight: 'bold',
    color: 'white',
    flexWrap: 'wrap',
  },
  titleButton: {
    flex: 1,
  },
});

interface TodoViewProps {
  todo: Todo;
  onTick:  (todo: Todo) => void;
  onDelete: (todo: Todo) => void;
  onPress: (todo: Todo) => void;
  isTicked: boolean;
}

const TodoView: React.FC<TodoViewProps> = ({
  todo,
  onPress,
  onDelete,
  onTick,
  isTicked = false
}) => {

  const onTickTodo = useCallback(() => {
    onTick?.(todo);
  }, [onTick, todo]);

  const onPressTodo = useCallback(() => {
    onPress?.(todo);
  }, [onPress, todo]);

  const onPressDelete = useCallback(() => {
    onDelete?.(todo);
  }, [onDelete, todo]);

  return (
    <View style={styles.container} key={todo.id}>
      <Icon
        testID='leftIcon'
        name={isTicked ? 'check-square' : 'square'}
        size={30}
        color="#900"
        style={styles.leftIcon}
        onPress={onTickTodo}
      />
      <TouchableOpacity style={styles.titleButton} onPress={onPressTodo}>
        <Text style={styles.title}>{todo.title}</Text>
      </TouchableOpacity>
      <Icon
        testID='rightIcon'
        name='trash-2'
        size={30}
        color="#900"
        style={styles.rightIcon}
        onPress={onPressDelete}
      />
    </View>
  );
};

export default TodoView;
