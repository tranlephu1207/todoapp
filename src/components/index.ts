export { default as TodoView } from './TodoView';