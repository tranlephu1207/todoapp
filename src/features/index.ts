export * from './todo';
export * from './authentication';