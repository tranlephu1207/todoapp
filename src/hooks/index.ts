export { default as useCachedResources } from './useCachedResources';
export { default as useAppState } from './useAppState';