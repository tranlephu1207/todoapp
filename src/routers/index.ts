export * from './types.d';
export { default as ScreenNames } from './enums';
export { default as MainRouter } from './mainRouter';
export { default as Navigator } from './navigator';