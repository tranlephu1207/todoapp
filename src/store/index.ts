import { applyMiddleware, compose, createStore } from 'redux';

import AsyncStorage from '@react-native-async-storage/async-storage';
import createSagaMdiddleware from '@redux-saga/core';
import persistCombineReducers from 'redux-persist/es/persistCombineReducers';
import persistStore from 'redux-persist/es/persistStore';
import rootReducer from './reducer';
import rootSaga from './saga';

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  blacklist: ['authentication']
};

const persistedReducer = persistCombineReducers(persistConfig, rootReducer);

const sagaMiddleware = createSagaMdiddleware();

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware))
);

export const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);

