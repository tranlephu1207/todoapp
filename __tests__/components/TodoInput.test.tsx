import Icon from '@expo/vector-icons/Feather';
import React from 'react';
import { TextInput } from 'react-native';
import { TodoInput } from '@root/src/components';
import renderer from 'react-test-renderer';

describe('TodoInput', () => {
  const icon: React.ReactElement<typeof Icon> = (<Icon testID='testIcon' name="plus" size={30} color="#900" />);
  const tree = renderer
    .create(
      <TodoInput
        value="test"
        onPressButton={() => console.log('successful onPressButton')}
        onChangeText={(txt) => console.log(txt)}
        buttonIcon={icon}
      />
    );

  it('should render correctly', async () => {
    const treeJSON = tree.toJSON();
    expect(treeJSON).toMatchSnapshot();
  });

  it('has correct values', async () => {    
    expect(tree.root.findByType(TextInput).props.value).toBe('test');
    expect(tree.root.findByProps({ testID: 'testIcon' }).type).toEqual(icon.type);
  });
});