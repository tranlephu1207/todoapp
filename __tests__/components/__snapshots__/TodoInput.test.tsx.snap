// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "rgb(222, 222, 222)",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "paddingBottom": 5,
      "paddingHorizontal": 16,
      "width": 750,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Do it now!"
    placeholderTextColor="white"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "white",
        "flex": 1,
        "flexShrink": 1,
        "fontSize": 22,
        "fontWeight": "bold",
        "maxHeight": 375,
      }
    }
    underlineColorAndroid="transparent"
    value="test"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 15,
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;

exports[`TodoInput should render correctly with Shallow 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "rgb(222, 222, 222)",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "paddingBottom": 5,
      "paddingHorizontal": 16,
      "width": 750,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Do it now!"
    placeholderTextColor="white"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "white",
        "flex": 1,
        "flexShrink": 1,
        "fontSize": 22,
        "fontWeight": "bold",
        "maxHeight": 375,
      }
    }
    underlineColorAndroid="transparent"
    value="test"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 15,
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
